hamilflow/models/particle_in_box.py

```diff
@@ -0,0 +1,143 @@
+from functools import cached_property
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import scipy as sp
+from pydantic import BaseModel, Field, computed_field, field_validator
+
+
+PLANCK_CONST = 6.626e-34
+PI = 3.142
+
+
+class FiniteBox(BaseModel):
+    r"""Definition of the potential of the one dimensional box
+
+    For consistency, we always use $\mathbf x$ for displacement,
+    $L$ for box size.
+
+    The potential distribution we use is
+
+    $$
+    V(x)=\left\{\begin{array}{ll}
+    -V_{0} & |x|<L \\
+    0 & |x|>L
+    \end{array}\right.
+    $$
+
+    :cvar length: length of the box
+    :cvar p_height: potential well height
+    """
+
+    length: float = Field(ge=0)
+    p_height: float = Field(g=0, default=1.0)
+
+
+class QuantumParticle(BaseModel):
+    r"""Definition of the quantum particle
+
+    :cvar energy: energy of the particle
+    :cvar mass: weight of the particle
+    """
+
+    energy: float = Field(l=0, ge=-FiniteBox.p_height)
+    mass: float = Field(ge=0, default=1e-5)
+
+
+class WaveFunctionConst(BaseModel):
+    r"""Schrodinger equation simplification terms"""
+
+    @property
```

Comment:
Preferably `cached_property` or some kinds of computed field? @emptymalei

We also have `hbar` in https://docs.scipy.org/doc/scipy/reference/constants.html#physical-constants

Reply:
> Preferably `cached_property` or some kinds of computed field? @emptymalei
> 
> We also have `hbar` in https://docs.scipy.org/doc/scipy/reference/constants.html#physical-constants

+1

1. To serialize the properties properly in Pydantic, we need `computed_field`.
2. Also better to cache it as this is used n-time-step times.